### ISSUE TITLE
WinAppDriver 0.7 - Run Elevated Fails with status 13 value error - "An unknown error occurred in the remote end while processing the command."

### ISSUE THREAD

**Issue by ianceicys** [Microsoft/WinAppDriver#104]
![image](https://cloud.githubusercontent.com/assets/3976990/20902349/c20cf64c-bb04-11e6-8479-64fdebd3160d.png)

repo of the issue: 
[![Video walkthrough](http://i.imgur.com/2tuojUw.png)](https://www.youtube.com/watch?v=IIELGWuWUhk)
 
Trying to write an automated test that interacts with Microsoft Word 2016 and I am receiving the following error: 

==========================================
POST /session HTTP/1.1
Accept: application/json, image/png
Connection: Keep-Alive
Content-Length: 125
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723

{"desiredCapabilities":{"app":"C:\\Program Files (x86)\\Microsoft Office\\root\\Office16\\WINWORD.EXE","platformName":"iOS"}}
SessionManager - Creating session
SessionManager - WinAppDriver succeeded loading MitaBroker
HTTP/1.1 500 Internal Error
Content-Length: 133
Content-Type: application/json

{"status":13,"value":{"error":"unknown error","message":"An unknown error occurred in the remote end while processing the command."}}

Source : https://github.com/ianceicys/WinAppDriver/blob/89c6726c891f3efc9af3dd4dc31cc238a20a2d70/Samples/C%23/WordTest/Scenario.cs

**Comment by sproatba** [comment]
I'm seeing this same thing, but my application isn't in a protected folder(C:\TeklaStructures).  I'm running as admin because visual studio is on my local machine and the application i'm testing is on a vm.  The problem is intermittent for me.  Works for a while then quits, then works for a while.  Can't seem to track down how to get it to work/not work consistently.

**Comment by ianceicys** [comment]
@sproatba One thing I've seen with intermittent failures is having to close and restart the WinAppDriver.exe process ("C:\Program Files (x86)\Windows Application Driver\WinAppDriver.exe") at the beginning and end  of each test. I seem to get more reliable failures (at least the same failure when I have a new process at the beginning of each test).

**Comment by viji-123** [comment]
I am also facing the same issue. Am trying to automate a wpf application. Tests are failing most of the times because of this issue. On an average out of 10 attempts it fails 7-8 times. Having hard time to run the tests. Please let me know if there is a fix or workaround for this issue.

**Comment by viji-123** [comment]
Can we expect the fix for this issue in the next release? This is a major issue we are facing here. Not able to run the tests because of this issue. 
Request to fix this issue ASAP

**Comment by viji-123** [comment]
Hi team,
Just curious to know when is the next release. Expecting the fix for this issue in the next release .

 I have done some tests using Winappdriver in our application. My team is in the deciding phase. I have to give a presentation of the working code. But the tests are failing because of this issue. Request the team to kindly provide the fix in the next release asap.

**Comment by hakkori** [comment]
Have you tried opening WinAppDriver as admin?

**Comment by tylerrrrrr** [comment]
I see the same issue. I'm trying to switch from my application to Excel to do a quick verification and then back to my application. Load ing the driver with Excel works only some of the times, very inconsistent. I have also tried running WInAppDriver in admin too.

Heres a typical attempt to start up Excel:
`POST /session HTTP/1.1
Accept-Encoding: gzip,deflate
Connection: Keep-Alive
Content-Length: 126
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4733
User-Agent: Apache-HttpClient/4.5.2 (Java/1.8.0_101)

{"desiredCapabilities":{"app":"C:\\Program Files (x86)\\Microsoft Office\\root\\Office16\\POWERPNT.exe","platformName":"iOS"}}
SessionManager - Creating session
SessionManager - WinAppDriver succeeded loading MitaBroker
HTTP/1.1 500 Internal Error
Content-Length: 133
Content-Type: application/json

{"status":13,"value":{"error":"unknown error","message":"An unknown error occurred in the remote end while processing the command."}}
`

**Comment by yodurr** [comment]
I haven't been able to repro this using the following apps.
WordAppId = @"C:\Program Files\Microsoft Office\Office16\WINWORD.exe";
WordAppId = @"C:\Program Files (x86)\WinAppDriverAdminTest\wpfnotepad.exe"; 

I'm using Windows Creators update insider release of Windows, and Visual Studio 2015 w/Update 3.

What Windows OS and Visual Studio version are people using?

**Comment by tylerrrrrr** [comment]
@yodurr 
Application: C:\\Program Files (x86)\\Microsoft Office\\root\\Office16\\POWERPNT.exe
OS: Windows 10 Enterprise. Version: 1511. OS build: 10586.753

Also we are not using Visual Studios for development, we use Java & Eclipse for example.

**Comment by viji-123** [comment]
@yodurr 
Am using visual studio for developing the tests. Below are the details
Windows - Microsoft Windows [Version 10.0.14393]
Visual studio Community 2015 , Version 14.0.25431.01 Update 3

 Am still getting the below error as I run the tests.

**{"status":13,"value":{"error":"unknown error","message":"An unknown error occurred in the remote end while processing the command."}}"**

Is there a way to check the logs for this error. Because all we can see is the above error in the Winappdriver.  I am working in a wpf application ,it takes 2-3 seconds to load , is that a reason for this issue? 

**Note**: I ran Winappdriver as admin , still it didnt help. The tests failed with this error

**Comment by viji-123** [comment]
Please let us know when we can expect a fix for this issue.

**Comment by yodurr** [comment]
Ok, found one bug and we'll get a fix out in the next release (asap).  The uppercase ".EXE" is currently not getting identified correctly and we'll fix this so you can use .EXE.  In the meantime only use all lowercase ".exe" in the app path.  

I don't believe this is the root cause for all issues on this thread so am not suggesting this bug is fixed, but hopefully this info helps some of you.

**Comment by viji-123** [comment]
Will this fix solve the "unknown error " issue ? If not please let us know when the "unknown error" issue will be fixed, so that we can plan accordingly. This "unknown error "issue is a blocker

**Comment by yodurr** [comment]
it solves the unknown error issue if you are using .EXE in your app path.  
Based on comments in this thread I think there's another bug that people are hitting and I don't have a status update on that yet.

**Comment by yodurr** [comment]
@tylerrrrrr do you have a lot of other applications running, are there a lot of other windows up?  That can slow down WinAppDriver's ability to find the Windows you're looking for.

Also especially since you're doing multiple session creation in your test it would help me set up a repro if I can see more of your test code.

**Comment by tylerrrrrr** [comment]
@yodurr I have my Windows Modern app, WinAppDriver, a file explorer, a running batch file, and 2 instances of Powerpoint(since my app launches it, and WAD launches a new instance of it. Have not found a way to hook into an already existing instance of an app).

What the basic flow is:
1) Launch and Interacting with my WM app(calculator for example)
`DesiredCapabilities capabilities = new DesiredCapabilities();
capabilities.setCapability("app", "MyApp");
generalDriver = new IOSDriver(new URL("http://127.0.0.1:4723"), capabilities);`
2) Click powerpoint within my app which launches ppt
3) Create new driver for ppt 
`DesiredCapabilities capabilities = new DesiredCapabilities();
capabilities.setCapability("app", "C:\\Program Files (x86)\\Microsoft Office\\root\\Office16\\POWERPNT.exe");
generalDriver = new IOSDriver(new URL("http://127.0.0.1:4723"), capabilities);`
4) Attempt to verify ppt opened(error here)

**Comment by yodurr** [comment]
@tylerrrrrr I think after #2 you should instead do the following for #3 to create a test session with the powerpoint window your app launched.  I tried this and it found the Powerpoint Window and got a screen shot of it.

`
// Create Desktop session to find powerpoint window launched by your app
DesiredCapabilities desktopCapabilities = new DesiredCapabilities();
desktopCapabilities.SetCapability("app", "Root");
desktopCapabilities.SetCapability("deviceName", "WindowsPC");
WindowsDriver<WindowsElement> DesktopSession = new WindowsDriver<WindowsElement>(new Uri(AppDriverUrl), desktopCapabilities);
Assert.IsNotNull(DesktopSession);
`

`
// Find powerpoint window
WindowsElement we = DesktopSession.FindElementByName("Presentation1 - PowerPoint");
// Take a screenshot of it
Screenshot we2 = we.GetScreenshot();
we2.SaveAsFile("c:\\temp\\ppt.jpg", System.Drawing.Imaging.ImageFormat.Jpeg);
`

**Comment by JanneRantala** [comment]
@yodurr I started getting same error in one of my build machines. 

{"status":13,"value":{"error":"unknown error","message":"An unknown error occurred in the remote end while processing the command."}}

You mentioned on February 3rd that you'll try to get a fix for next release, is that already available? Is that fix for winappdriver.exe?

**Comment by eddiekorbz** [comment]
First time using Appium for windows and such a bummer to immediately run into this issue. So I am able to start my app which opens up showing a login window. It then immediately errors out with error in this thread. 
==========================================
POST /session HTTP/1.1
Accept: application/json, image/png
Connection: Keep-Alive
Content-Length: 145
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723

{"desiredCapabilities":{"deviceName":"WindowsPC","app":"C:\\Program Files\\Symed\\OneAppPro\\Symed.Framework.Stub.exe","platformName":"Windows"}}
SessionManager - Creating session
SessionManager - WinAppDriver succeeded loading MitaBroker
HTTP/1.1 500 Internal Error
Content-Length: 133
Content-Type: application/json

{"status":13,"value":{"error":"unknown error","message":"An unknown error occurred in the remote end while processing the command."}}

**Comment by eddiekorbz** [comment]
Here is my code:
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium.Appium.Windows;
using OpenQA.Selenium.Remote;

namespace ThickClient
{
    [TestClass]
    public class OneAppBase
    {
        protected const string AppDriverUrl = "http://127.0.0.1:4723"; 
        protected static WindowsDriver<WindowsElement> OneAppSession;

        //protected const string PasswordValue = "password";
        [ClassInitialize] 
        public static void OneAppBaseSetup(TestContext testcontext)
        {
            if (OneAppSession == null)
            {
                //launch the Notepad app 
                DesiredCapabilities appCapabilities = new DesiredCapabilities();

                appCapabilities.SetCapability("deviceName", "WindowsPC");
                appCapabilities.SetCapability("app", @"C:\Program Files\Symed\OneAppPro\Symed.Framework.Stub.exe");
                OneAppSession = new WindowsDriver<WindowsElement>(new Uri(AppDriverUrl), appCapabilities);
                Assert.IsNotNull(OneAppSession);

            }
        }

        [TestMethod]
        public void Login()
        {
            //Login to OneApp
            OneAppSession.FindElementByName("Password:").SendKeys("password");
            Assert.IsNotNull(OneAppSession);

        }

        //[ClassCleanup] 
        //public static void NotepadBaseTearDown()
        //{
        //    if (OneAppSession != null) 
        //    { 
        //         OneAppSession.Dispose(); 
        //         OneAppSession = null; 
        //    } 
        //} 

    }
}

**Comment by caulfield2** [comment]
I too am running in to this issue with non-Windows applications. Has a workaround or solution been identified?

**Comment by Manvibha** [comment]
I too had the same issue and workaround was "Run WinAppDriver.exe as an administrator". 
It Just worked !!!

~Mahi

**Comment by jhmyq2** [comment]
I am just trying to run through the C# calculator example and am getting this error.  I am using Visual Studio 2017, Windows 10 version 1607 build 14393.  I tried the code that came in the example and was getting a different unknown error.  I then switched the app to @"C:\Windows\System32\calc.exe" and the calculator will start but then I get the following:

SessionManager - Creating session
SessionManager - WinAppDriver succeeded loading MitaBroker
HTTP/1.1 500 Internal Error
Content-Length: 133
Content-Type: application/json

{"status":13,"value":{"error":"unknown error","message":"An unknown error occurred in the remote end while processing the command."}}


This is the code that causes it.  Am I doing anything wrong here?

DesiredCapabilities appCapabilities = new DesiredCapabilities();
                appCapabilities.SetCapability("app", @"C:\Windows\System32\calc.exe");
                appCapabilities.SetCapability("deviceName", "WindowsPC");
                CalculatorSession = new WindowsDriver<WindowsElement>(new Uri(WindowsApplicationDriverUrl), appCapabilities);

**Comment by Slesa** [comment]
I have the same problem, too. But neither does it help to start WinAppDriver as Admin, nor do I have a lower cased .exe in file name. I also ensured that my app is loaded (with MessageBoxes - the hard way ;)).

So, what else could it be? Would be great to have the source code of WinAppDriver, then I could debug. Maybe then we also could get the missing arguments working...

**Comment by yodurr** [comment]
Better error messaging is coming in v1.0 and hopefully we can learn more about where the issue is with that release.

**Comment by timotiusmargo** [comment]
As of last week, we have released Windows Application Driver [v1.0-RC](https://github.com/Microsoft/WinAppDriver/releases/tag/v1.0-RC). This version is not only more robust and stable, but it also returns more meaningful exceptions in the case of session creation error.

Many applications have unique launch sequences that do not fall into the generic application startup WinAppDriver can handle. In such case, WinAppDriver would return a **failure to locate application main window**. In this latest version however, we introduced a new [Attaching to an Existing App Window](https://github.com/Microsoft/WinAppDriver/blob/v1.0-RC/README.md#attaching-to-an-existing-app-window) that can be used in combination with [Desktop Session](https://github.com/Microsoft/WinAppDriver/blob/v1.0-RC/README.md#creating-a-desktop-session) to create a session from any successfully launched top level application window.

As this issue is now getting broader, would you please:
1. Download and install the latest WinAppDriver version
2. Re-validate your scenario you posted above
3. Attempt the [Attaching to an Existing App Window](https://github.com/Microsoft/WinAppDriver/blob/v1.0-RC/README.md#attaching-to-an-existing-app-window) approach if application is successfully launched but the session creation failed to locate the window.

If you still encounter the same issue only when running `WinAppDriver` **as administrator** (elevated), then post the exact repro steps and logs in this thread. Otherwise, please create a new issue with your findings.

**Comment by ewancombe** [comment]
Timotius,

I have been having a number of issues but I think they all actually stem from this issue. The WinAppDriver.exe stopps immediately when I launch it. Having never used it before I assumed that was normal but I watched your video carefully and I saw it should run and should say the ip and port it is listening.

As requested, here are my steps, they are actually just the basic steps from a blog on getting started with this stuff:

1.Download Windows Application Driver installer from https://github.com/Microsoft/WinAppDriver/releases

2.Run the installer on a Windows 10 machine where your application under test is installed and will be tested

3.Run WinAppDriver.exe from the installation directory (E.g. C:\Program Files (x86)\Windows Application Driver)

WinAppDriver opens a CMD window which then closes within half a second. I believe that's my issue.

Do you have any advice on this?

**Comment by ewancombe** [comment]
Oh I have tried running it as administrator too.

**Comment by khouzam** [comment]
Hi @ewancombe,

In the future it is better to open a new issue instead of using an older closed one.

Two things for you to try.

1. Make sure to enable developer mode from the settings.
2. Open cmd and launch WinAppDriver from there. This will at least keep the window open so you can see the error message that comes up.

Thanks

**Comment by ewancombe** [comment]
> Hi @ewancombe,
> 
> In the future it is better to open a new issue instead of using an older closed one.
> 
> Two things for you to try.
> 
> 1. Make sure to enable developer mode from the settings.
> 2. Open cmd and launch WinAppDriver from there. This will at least keep the window open so you can see the error message that comes up.
> 
> Thanks

khouzan,

Thanks I set Developer Mode to On and that definitely helped. The Command Window stays open now. Now I get "OpenQA.Selenium.WebDriverException: 'Failed to locate opened application window with appId: C:\Windows\System32\calc.exe, and processId: 19896'". So you have helped and I'm 1 step further along...

Ewan

**Comment by hassanuz** [comment]
Thanks @ewancombe, we're marking this issue as closed. We'll continue to follow up with you [here](https://github.com/Microsoft/WinAppDriver/issues/516).

**Comment by erichui190125** [comment]
Hi guys, got stuck on this for a little while. Run winAppDriver.exe in admin mode and it should work! At least it did for me. Happy testing.

**Comment by ravi46kom** [comment]
Hi @erichui190125 @hassanuz 
It's not working in my condition I tried a lot, i upgraded a project .net 4.7 to core and then i'm geting flaky errors using appium.webdriver. 
<img width="488" alt="image" src="https://github.com/microsoft/WinAppDriver/assets/74744300/4f5b7d78-c0a9-49d1-ae41-07ebcf706a24">

**Comment by ajimiathul** [comment]
Any update on this issue?? Still facing the same problem. Tried Run as administrator but no luck.
